File: vastbase/src/test/java/com/dbx/agent/vastbase/VastbaseAgentTest.java
package com.dbx.agent.vastbase;

import com.dbx.agent.DatabaseAgent;
import com.dbx.agent.test.JdbcFakeExecutionBehaviorTest;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class VastbaseAgentTest extends JdbcFakeExecutionBehaviorTest {
    @Override
    protected DatabaseAgent createAgent() {
        return new VastbaseAgent();
    }

    @Override
    protected String resultSetSql() {
        return "CALL sample_proc()";
    }

    @Test
    void declaresVastbasePostgresLikeProfile() {
        VastbaseAgent agent = new VastbaseAgent();

        Assertions.assertEquals("cn.com.vastbase.Driver", agent.getProfile().getDriverClass());
        Assertions.assertEquals("jdbc:vastbase://{host}:{port}/{database}", agent.getProfile().getUrlTemplate());
    }
}

File: vastbase/src/test/kotlin/com/dbx/agent/vastbase/VastbaseAgentTest.kt
package com.dbx.agent.vastbase

import com.dbx.agent.DatabaseAgent
import com.dbx.agent.test.JdbcFakeExecutionBehaviorTest
import kotlin.test.Test
import kotlin.test.assertEquals

class VastbaseAgentTest : JdbcFakeExecutionBehaviorTest() {
    override fun createAgent(): DatabaseAgent {
        return VastbaseAgent()
    }

    override fun resultSetSql(): String = "CALL sample_proc()"

    @Test
    fun `declares vastbase postgres-like profile`() {
        val agent = VastbaseAgent()

        assertEquals("cn.com.vastbase.Driver", agent.profile.driverClass)
        assertEquals("jdbc:vastbase://{host}:{port}/{database}", agent.profile.urlTemplate)
    }
}
